### PR TITLE
Serialize knownnodes to json

### PR DESCRIPTION
**Pull request by g1itch**
Hello!

I've implemented serialization of `knownnodes` to json (#1232).

### CODE REVIEW

**Comment by g1itch** [comment]
Yes, it seems I have no access. Hope in this case it's not a problem because this is a trivial changes. It took a couple of hours or so to write.

Can you also remind me which sentence I should write in commit comment "fixes #1232"? I'll reword.

**Comment by PeterSurda** [comment]
It's not a problem in this case, it's just for the future as more people get involved it makes it more complicated for me to manage.

Regarding referencing, it's described in github documentation: https://blog.github.com/2013-01-22-closing-issues-via-commit-messages/

**Comment by g1itch** [comment]
Hmm, I understand why it wont work like I thought. `master` still is default branch in this repo. The article says 

> once the commit is merged into your default branch

**Comment by PeterSurda** [comment]
I think I reset the default branch to v0.6, no? It will only close the issue once I merge the PR, not when you open it.

**Comment by g1itch** [comment]
OK, I didn't really tested yet. I'll try to add that sentence to last commit's comment if this PR will be considered acceptable.

File: src/knownnodes.py [hunk 1]
```diff
@@ -1,49 +1,131 @@
...
 
 knownNodesTrimAmount = 2000
 
 # forget a node after rating is this low
 knownNodesForgetRating = -0.5
 
-def saveKnownNodes(dirName = None):
+
+def json_serialize_knownnodes(output):
+    _serialized = []
```
Comment (omkar1117) [inline src/knownnodes.py]: better to add doc string.

File: src/knownnodes.py [hunk 2]
```diff
@@ -1,49 +1,131 @@
...
     with knownNodesLock:
         with open(os.path.join(dirName, 'knownnodes.dat'), 'wb') as output:
-            pickle.dump(knownNodes, output)
+            json_serialize_knownnodes(output)
+
+
+def addKnownNode(stream, peer, lastseen=None, is_self=False):
+    if lastseen is None:
+        lastseen = time.time()
+    knownNodes[stream][peer] = {
```
Comment (omkar1117) [inline src/knownnodes.py]: "lastseen" : lastseen if lastseen else time.time()

**Comment by f97ada87** [comment]
This worked well for me (v0.6 current on Linux). Good work. 

However, if the JSON file is corrupted (due to any of a number of factors, including human error), `knownnodes.readKnownNodes()` fallbacks to `pickle_deserialize_old_knownnodes()` and crashes hard. Some better error handling might be helpful in such scenarios. 

For example, you could issue a _meaningful_ fatal error message and exit cleanly. Or even better, issue a warning, rename the corrupt/unreadable file to a bak extension, start a fresh one and continue the startup sequence seamlessly.

File: src/knownnodes.py (hunk 2 above)
Comment (PeterSurda) [reply src/knownnodes.py]: :+1:

**Comment by g1itch** [comment]
@f97ada87, pybitmessage can start without knownnodes.dat. So I think it will be better just to handle the exception.

**Comment by f97ada87** [comment]
Agreed - it's not a condition severe enough to justify a quit, soft or hard.

**Pull request merged**